### PR TITLE
Pipeline build implementation

### PR DESCRIPTION
**Pull request by albertob13**
Pipeline build first implementation with app signing management.

### CODE REVIEW

File: .github/workflows/pipeline-build.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+name: build-action
```
Comment (diegorondini) [inline .github/workflows/pipeline-build.yml:1]: Leave the pipeline stuff as a separate PR (just leave this one open, we'll eventually rebase).
We first need to merge the keys refactoring. Please open a dedicated PR for keys refactoring with the changes requested here.

File: .github/workflows/pipeline-build.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        uses: gradle/gradle-build-action@v2
+
+      - name: Override Android SDK
+        run: ./install_android-hidden-api-jar.sh
+
+      - name: Gradle build
+        env:
+          PASS_KEYSTORE_KYNETICS: ${{ secrets.PASS_KEYSTORE_KYNETICS }}
+          PASS_KEY_KYNETICS: ${{ secrets.PASS_KEY_KYNETICS }}
+          ALIAS_KEY_KYNETICS: ${{ secrets.ALIAS_KYNEICS_KEY }}
```
Comment (diegorondini) [inline .github/workflows/pipeline-build.yml:28]: Typo in var name `KYNEICS`.

File: .github/workflows/pipeline-build.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+      - name: Setup gradle
+        uses: gradle/gradle-build-action@v2
+
+      - name: Override Android SDK
+        run: ./install_android-hidden-api-jar.sh
+
+      - name: Gradle build
+        env:
+          PASS_KEYSTORE_KYNETICS: ${{ secrets.PASS_KEYSTORE_KYNETICS }}
```
Comment (diegorondini) [inline .github/workflows/pipeline-build.yml:26]: Please use English language order for variables naming. If we're talking about the Kynetics keystore password, the variable should be named `KYNETICS_KEYSTORE_PASS`.

File: keystore/buildkeystore.sh [hunk 4]
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+#Script used for generating public.jks keystore containing aosp and nxp platform keys.
+#Delete current 'public.jks' and make sure ./keystore/ is free of subdirectories before run it.
```
Comment (diegorondini) [inline keystore/buildkeystore.sh:4]: > free of subdirectories before run it

This is not true. The `keystore/` folder should include the subfolders with the `.pk8` and `x509.pem` key pairs.

File: keystore/buildkeystore.sh [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+BASE_PWD=$PWD
+
+for d in */ ; do
+	cd $d
+	rm -f ./$(basename $d)_decrypted_key.p12
```
Comment (diegorondini) [inline keystore/buildkeystore.sh:10]: The `.p12` is not "decrypted". It's just a different (combined) format to store the keys:
https://support.ksoftware.net/support/solutions/articles/17251-what-is-a-p12-file-or-a-pkcs12-file-

File: keystore/buildkeystore.sh [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+	openssl pkcs8 -inform DER -nocrypt -in "platform.pk8" | openssl pkcs12 -export -in "platform.x509.pem" -inkey /dev/stdin -name "platform_$(basename $d)" -password pass:$(basename $d) -out "$(basename $d)_decrypted_key.p12"
+	keytool  -importkeystore  -deststorepass keystore -srckeystore "$(basename $d)_decrypted_key.p12" -srcstoretype PKCS12 -srcstorepass $(basename $d) -destkeystore "$BASE_PWD/public.jks"
+	cd $BASE_PWD
+done
+
+echo "Keystore entries: " 
+keytool -list -keystore public.jks -storepass keystore
+
+echo ""
+if [ $? -eq 0 ];
```
Comment (diegorondini) [inline keystore/buildkeystore.sh:20]: This `if` will always be true as you're just checking the return value of the previous command, which in this case is `echo ""`.

File: uf-client-ui-example/build.gradle [hunk 7]
```diff
@@ -31,16 +33,25 @@ android {
         main.java.srcDirs += 'src/main/kotlin'
     }
 
-    def isRunningOnTravis = System.getenv("CI") == "true"
+    def passKeyStoreKynetics = System.getenv("PASS_KEYSTORE_KYNETICS")
```
Comment (diegorondini) [inline uf-client-ui-example/build.gradle:36]: See the other comment about English language order. This should have been: `def KyneticsKeystorePass = System.getenv("KYNETICS_KEYSTORE_PASS")`

File: uf-client-ui-example/build.gradle [hunk 8]
```diff
@@ -31,16 +33,25 @@ android {
...
+    def aliasKyneticsKey = System.getenv("ALIAS_KYNETICS_KEY")
+    def passKeyKynetics = System.getenv("PASS_KEY_KYNETICS")
```
Comment (diegorondini) [inline uf-client-ui-example/build.gradle:38]: Same as above

**Review by diegorondini — CHANGES_REQUESTED** [review]

File: keystore/buildkeystore.sh (hunk 4 above)
Comment (albertob13) [reply keystore/buildkeystore.sh:4]: I meant that any other subfolders not containing the necessary keys will not make the script work properly.

**Review by diegorondini — CHANGES_REQUESTED** [review]

**Review by danielesergio — APPROVED** [review]

**Pull request merged**